rust/src/ctypes.rs

```diff
@@ -0,0 +1,33 @@
+use ffi_utils::strings::{
+    string_to_c_char
+};
+
+use std;
+use std::os::raw::c_char;
+
+use errors;
+use Toodle;
+
+pub struct ResultC<T> {
+    pub value: *const T,
+    pub error: *const c_char
+}
+
+impl From<std::result::Result<Toodle, errors::Error>> for ResultC<Toodle> {
+    fn from(result: std::result::Result<Toodle, errors::Error>) -> Self {
+        match result {
+            Ok(val) => {
+                ResultC::<Toodle> {
+                    value: Box::into_raw(Box::new(val)),
+                    error: string_to_c_char(String::from(""))
```

Comment:
Can we make `error` a `null_mut` rather than an empty string? Checking for the presence or absence of something is much better than checking for whether a string is empty or not.

Reply:
Also, nit: `String::new()` is the prefered way of getting an empty `String`